### PR TITLE
Add script to convert a directory of glsl-fuzz shaders into a form suitable for use with spirv-fuzz.

### PR DESCRIPTION
**Pull request by afd**
In the process refactored runspv a bit in order to reuse parts of it.

### CODE REVIEW

File: python/src/main/python/drivers/runspv.py [hunk 1]
```diff
@@ -327,21 +327,37 @@ def remove_end(str_in: str, str_end: str):
     return str_in[:-len(str_end)]
 
 
+# Runs spirv-opt on |spv_file| with arguments |spirv_opt_args|.  The optimized binary file is
```
Comment (paulthomson) [inline python/src/main/python/drivers/runspv.py]: It is more common to use docstrings: """ """ within the function body.

File: python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py [hunk 2]
```diff
@@ -79,9 +83,9 @@ def main() -> None:
                 constantWord=[int_representation])
             fact_list.append(dict(constantUniformFact=fact_constant_uniform))
 
-    with open(os.path.splitext(args.shader_job)[0] + ".facts", "w") as f:
+    with open(args.output_file, "w") as f:
```
Comment (paulthomson) [inline python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py]: You didn't really change this, but it is best to use `encoding='utf-8', errors='ignore'`, or just use `runspv.open_helper`

File: python/src/main/python/drivers/runspv.py (hunk 1 above)
Comment (afd) [reply python/src/main/python/drivers/runspv.py]: Done.

File: python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py (hunk 2 above)
Comment (afd) [reply python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py]: Done (the latter)

File: python/src/main/python/drivers/runspv.py [hunk 3]
```diff
@@ -329,19 +329,40 @@ def remove_end(str_in: str, str_end: str):
...
 
-    cmd = [spirvopt_path(), spv_file, '-o', result]
+    cmd = [spirvopt_path(), spv_file, '-o', output]
     cmd += spirv_opt_args
 
     subprocess_helper(cmd, timeout=TIMEOUT_SPIRV_OPT_SECONDS)
 
-    return result
+
+# Runs glslangValidator on |glsl_shader| to convert it to SPIR-V.  The resulting SPIR-V file is
```
Comment (paulthomson) [inline python/src/main/python/drivers/runspv.py]: Here as well.

File: python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py [hunk 4]
```diff
@@ -16,24 +16,28 @@
...
 import json
-import os
 import struct
+import sys
 from typing import Any, Dict, List
 
 import runspv
 
 
 # Turns a GraphicsFuzz .json file into a spirv-fuzz .facts file, with one fact per word of uniform
```
Comment (paulthomson) [inline python/src/main/python/drivers/shader_job_uniforms_to_spirv_fuzz_facts.py]: Could fix this as well

File: python/src/main/python/drivers/runspv.py (hunk 3 above)
Comment (afd) [reply python/src/main/python/drivers/runspv.py]: Done.

**Pull request merged**